java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -0,0 +1,519 @@
+        }
+        float8Vector.setSafe(rowCount, holder);
+        float8Vector.setValueCount(rowCount + 1);
+    }
+
+    private static void updateVector(VarCharVector varcharVector, String value, boolean isNonNull, int rowCount) {
+        NullableVarCharHolder holder = new NullableVarCharHolder();
+        holder.isSet = isNonNull? 1: 0;
+        varcharVector.setIndexDefined(rowCount);
+        if (isNonNull) {
+            byte[] bytes = value.getBytes(StandardCharsets.UTF_8);
+            holder.buffer = varcharVector.getAllocator().buffer(bytes.length);
+            holder.buffer.setBytes(0, bytes, 0, bytes.length);
+            holder.start = 0;
+            holder.end = bytes.length;
+        } else {
+        	holder.buffer = varcharVector.getAllocator().buffer(0);
+        }
+        varcharVector.setSafe(rowCount, holder);
+        varcharVector.setValueCount(rowCount + 1);
+    }
+
+    private static void updateVector(DateMilliVector dateMilliVector, Date date, boolean isNonNull, int rowCount) {
+        NullableDateMilliHolder holder = new NullableDateMilliHolder();
+        holder.isSet = isNonNull? 1: 0;
+        if (isNonNull) {
+            holder.value = date.getTime();
+        }
+        dateMilliVector.setSafe(rowCount, holder);
+        dateMilliVector.setValueCount(rowCount + 1);
+    }
+
+    private static void updateVector(TimeMilliVector timeMilliVector, Time time, boolean isNonNull, int rowCount) {
+        NullableTimeMilliHolder holder = new NullableTimeMilliHolder();
+        holder.isSet = isNonNull? 1: 0;
+        if (isNonNull && time != null) {
+            holder.value = (int)time.getTime();
+        }
+        timeMilliVector.setSafe(rowCount, holder);
+        timeMilliVector.setValueCount(rowCount + 1);
+    }
+
+    private static void updateVector(TimeStampVector timeStampVector, Timestamp timestamp, boolean isNonNull, int rowCount) {
+        //TODO: Need to handle precision such as milli, micro, nano
+        timeStampVector.setValueCount(rowCount + 1);
+        if (timestamp != null) {
+            timeStampVector.setSafe(rowCount, timestamp.getTime());
+        } else {
+            timeStampVector.setNull(rowCount);
+        }
+    }
+
+    private static void updateVector(VarBinaryVector varBinaryVector, InputStream is, boolean isNonNull, int rowCount) throws IOException {
+        varBinaryVector.setValueCount(rowCount + 1);
+        if (isNonNull && is != null) {
+            VarBinaryHolder holder = new VarBinaryHolder();
+            ArrowBuf arrowBuf = varBinaryVector.getDataBuffer();
+            holder.start = 0;
+            byte[] bytes = new byte[DEFAULT_STREAM_BUFFER_SIZE];
+            int total = 0;
+            while (true) {
+                int read = is.read(bytes, 0, DEFAULT_STREAM_BUFFER_SIZE);
+                if (read == -1) {
+                    break;
+                }
+                arrowBuf.setBytes(total, new ByteArrayInputStream(bytes, 0, read), read);
```

Comment:
I think you don't need to wrap to a stream and that there's a method accepting an byte[] argument directly